.travis.yml

```diff
@@ -1,62 +1,103 @@
+sudo: false
 language: android
 
-android:
-  components:
-    - tools
-    - platform-tools
-    - build-tools-27.0.0
-    - android-16
-    - android-19
-    - android-21
-    - android-27
-    - extra-android-m2repository
-    - sys-img-armeabi-v7a-android-16
-    - sys-img-armeabi-v7a-android-19
-    - sys-img-armeabi-v7a-android-21
-
-before_cache:
-  - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
-  - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
-cache:
-  directories:
-    - $HOME/.gradle/caches/
-    - $HOME/.gradle/wrapper/
-    - $HOME/.android/build-cache
+jdk:
+- oraclejdk8
 
 env:
-  matrix:
-    - ANDROID_TARGET=android-16 ANDROID_ABI=armeabi-v7a
-    - ANDROID_TARGET=android-19 ANDROID_ABI=armeabi-v7a
-    - ANDROID_TARGET=android-21 ANDROID_ABI=armeabi-v7a
   global:
-    - ADB_INSTALL_TIMEOUT=5 # increase from default of 2 mins
+  - ADB_INSTALL_TIMEOUT=8 # increase from default of 2 mins
+  - ANDROID_EMULATOR=test # default emulator name (used in tests)
+  - ANDROID_ABI=armeabi-v7a
+  - ANDROID_TARGET=android-21
+  - NDK_VERSION=r16b
+  - EMULATOR_FLAVOR=default
+  - ANDROID_HOME=/usr/local/android-sdk
+  - TOOLS=${ANDROID_HOME}/tools
+  - PATH=${ANDROID_HOME}:${ANDROID_HOME}/emulator:${TOOLS}:${TOOLS}/bin:${ANDROID_HOME}/platform-tools:${PATH}
+  - QEMU_AUDIO_DRV=none
+
+android:
+  components:
+  # installing tools to start, then use `sdkmanager` below to get the rest
+  - tools
 
-# see https://github.com/travis-ci/travis-ci/issues/8874
 before_install:
-  - yes | sdkmanager "platforms;android-27"
+- bundle install
+- ./scripts/install-ndk.sh
+- env
+
+install:
+- echo y | sdkmanager "platform-tools" >/dev/null
```

Comment:
Why is this now run under `install` block rather than `android components`? Would this lead to duplicated effort?

Reply:
If you use the `components` version, you get an older version of the tools and don't have `avdmanager` and some images don't install.

Reply:
Does that also apply more generally to dependencies which are specifically versioned, such as build-tools etc?

Reply:
I can't say I fully understand it, but the duplication is specifically what makes this combination works. Trying to install via `components` instead leads to tools missing.